Source/WebKit/UIProcess/API/glib/WebKitImage.cpp

```diff
@@ -0,0 +1,476 @@
+    }
+}
+
+static void webkitImageGetProperty(GObject* object, guint propId, GValue* value, GParamSpec* paramSpec)
+{
+    auto* image = WEBKIT_IMAGE(object);
+
+    switch (propId) {
+    case PROP_WIDTH:
+        g_value_set_int(value, webkit_image_get_width(image));
+        break;
+    case PROP_HEIGHT:
+        g_value_set_int(value, webkit_image_get_height(image));
+        break;
+    case PROP_STRIDE:
+        g_value_set_uint(value, webkit_image_get_stride(image));
+        break;
+    default:
+        G_OBJECT_WARN_INVALID_PROPERTY_ID(object, propId, paramSpec);
+    }
+}
+
+static void webkit_image_class_init(WebKitImageClass* imageClass)
+{
+    GObjectClass* objectClass = G_OBJECT_CLASS(imageClass);
+    objectClass->set_property = webkitImageSetProperty;
+    objectClass->get_property = webkitImageGetProperty;
+
+    /**
+     * WebKitImage:width:
+     *
+     * The image width in pixels.
+     *
+     * Since: 2.52
+     */
+    sObjProperties[PROP_WIDTH] =
+    g_param_spec_int(
+        "width",
+        nullptr, nullptr,
+        1, G_MAXINT, 1,
+        static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
+
+    /**
+     * WebKitImage:height:
+     *
+     * The image height in pixels.
+     *
+     * Since: 2.52
+     */
+    sObjProperties[PROP_HEIGHT] =
+    g_param_spec_int(
+        "height",
+        nullptr, nullptr,
+        1, G_MAXINT, 1,
+        static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
+
+    /**
+     * WebKitImage:stride:
+     *
+     * The image stride, in bytes. This value indicates the amount of
+     * memory occupied by each row of pixels in the image. Note that
+     * the stride may be larger than the image width multiplied by the
+     * amount of bytes used to represent each pixel.
+     *
+     * Since: 2.52
+     */
+    sObjProperties[PROP_STRIDE] =
+    g_param_spec_uint(
+        "stride",
+        nullptr, nullptr,
+        RGBA8BytesPerPixel, G_MAXUINT, 4,
+        static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
+
+    g_object_class_install_properties(objectClass, N_PROPERTIES, sObjProperties.data());
+}
+
+// When instantiating this class and providing a CompletionHandler to instantiate the bytes, make sure
+// this uses #CompletionHandlerCallThread::AnyThread if you first instantiate the class bytes asynchronously.
+WebKitImage* webkitImageNew(int width, int height, guint stride, CompletionHandler<GRefPtr<GBytes>(WebKitImage*)>&& createBytesFn)
+{
+    RELEASE_ASSERT(createBytesFn);
+    RELEASE_ASSERT(static_cast<gsize>(width) * RGBA8BytesPerPixel <= stride);
+
+    auto* image = WEBKIT_IMAGE(g_object_new(WEBKIT_TYPE_IMAGE,
+        "width", width,
+        "height", height,
+        "stride", stride,
+        nullptr));
+    image->priv->createBytesFn = WTFMove(createBytesFn);
+
+    return image;
+}
+
+/**
+ * webkit_image_get_width:
```

Comment:
Conversely, let's annotate this to indicate which property this getter returns:


```suggestion
 * webkit_image_get_width: (get-property width):
```

(The same goes for the getter functions of the `height` and `stride` properties.)